classes/Http/Controller/TalkFormController.php

```diff
@@ -0,0 +1,149 @@
+    private $talkHelper;
+    private $callForPapers;
+
+    public function __construct(
+        Authentication $authentication,
+        TalkHelper $talkHelper,
+        CallForPapers $callForPapers,
+        Twig_Environment $twig,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->authentication = $authentication;
+        $this->talkHelper     = $talkHelper;
+        $this->callForPapers  = $callForPapers;
+
+        parent::__construct($twig, $urlGenerator);
+    }
+
+    public function editAction(Request $request)
+    {
+        $talkId = (int) $request->get('id');
+        // You can only edit talks while the CfP is open
+        // This will redirect to "view" the talk in a read-only template
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set(
+                'flash',
+                [
+                    'type'  => 'error',
+                    'short' => 'Read Only',
+                    'ext'   => 'You cannot edit talks once the call for papers has ended', ]
+            );
+
+            return new RedirectResponse($this->url('talk_view', ['id' => $talkId]));
+        }
+
+        if (empty($talkId)) {
+            return $this->redirectTo('dashboard');
+        }
+
+        $userId = $this->authentication->userId();
+
+        $talk = Talk::find($talkId);
+
+        if (!$talk instanceof Talk || (int) $talk['user_id'] !== $userId) {
+            return $this->redirectTo('dashboard');
+        }
+
+        $data = [
+            'formAction'     => $this->url('talk_update'),
+            'talkCategories' => $this->talkHelper->getTalkCategories(),
+            'talkTypes'      => $this->talkHelper->getTalkTypes(),
+            'talkLevels'     => $this->talkHelper->getTalkLevels(),
+            'id'             => $talkId,
+            'title'          => \html_entity_decode($talk['title']),
+            'description'    => \html_entity_decode($talk['description']),
+            'type'           => $talk['type'],
+            'level'          => $talk['level'],
+            'category'       => $talk['category'],
+            'desired'        => $talk['desired'],
+            'slides'         => $talk['slides'],
+            'other'          => $talk['other'],
+            'sponsor'        => $talk['sponsor'],
+            'buttonInfo'     => 'Update my talk!',
+        ];
+
+        return $this->render('talk/edit.twig', $data);
+    }
+
+    public function createAction(Request $request)
+    {
+        // You can only create talks while the CfP is open
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set(
+                'flash',
+                [
+                    'type'  => 'error',
+                    'short' => 'Error',
+                    'ext'   => 'You cannot create talks once the call for papers has ended', ]
+            );
+
+            return $this->redirectTo('dashboard');
+        }
+
+        $data = [
```

Comment:
While we are at it, can we inline `$data`?

Also see #851, where I already started working on that.